src/js/server/devpmt-ppt-dispatchers.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * Dispatchers for PPT pages and workflow
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.devpmt.ppt");
+
+gpii.devpmt.ppt.checkAuthorization = function (that, req, res /*, next */) {
+    if (req.session.loggedInToPPT) {
+        return true;
+    }
+    res.status("401").redirect("/");
+};
+
+/**
+ * Index Handler for /
+ */
+fluid.defaults("gpii.devpmt.dispatchers.index", {
+    gradeNames: "gpii.devpmt.baseDispatcher",
+    defaultTemplate: "index",
+    rules: {
+        contextToExpose: {
+        }
+    },
+    invokers: {
+        checkAuthorization: {
+            funcName: "gpii.devpmt.ppt.checkAuthorization",
+            args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // req, res, next
+        },
+        contextPromise: {
+            funcName: "gpii.devpmt.dispatchers.index.contextPromise",
+            args: ["{that}", "{devpmt}", "{arguments}.0"]
+        }
+    }
+});
+
+gpii.devpmt.dispatchers.index.contextPromise = function (that, devpmt, req) {
+    var promTogo = fluid.promise();
+    var prefsSafesProm = devpmt.prefsSafesListingDataSource.get();
+    prefsSafesProm.then(function (data) {
+        promTogo.resolve({
+            prefsSafesList: data.rows,
+            perms: req.session.perms
+        });
+    });
+    return promTogo;
+};
+
+// URLPATH /editprefs/:npset
+fluid.defaults("gpii.devpmt.ppt.loginHandler", {
+    gradeNames: ["gpii.devpmt.baseDispatcher"],
+    defaultTemplate: "ppt-login",
+    method: "use",
+    // list of extra permissions to be added to session on successful login
+    permissions: {
+    },
+    invokers: {
+        checkAuthorization: {
+            funcName: "gpii.devpmt.ppt.loginHandler.checkAuthorization",
+            args: ["{that}", "{gpii.devpmt}.gpiiExpressUserApi", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // req, res, next
+        }
+    }
+});
+
+gpii.devpmt.ppt.loginHandler.checkAuthorization = function (that, userAPI, req, res /*, next */) {
+    if (req.method !== "POST") {
+        return true;
+    }
+
+    if (req.body.username === "morphic" && req.body.password === "gpii") {
+        req.session.loggedInToPPT = true;
+        req.session.perms = that.options.permissions;
+        res.redirect("/ppt");
+    }
+    else {
+        res.redirect("/");
+    }
+
+    // During our next round of work, we will finish hooking the PPT up to
```

Comment:
Move this comment to appropriate place such as before the definition of the `gpii.devpmt.ppt.loginHandler` component. The following commented code can be removed if it's no longer in use. Thanks.

Reply:
Done and done